staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go

```diff
@@ -169,17 +174,60 @@ func GetValueFromIntOrPercent(intOrPercent *IntOrString, total int, roundUp bool
 	return value, nil
 }
 
+//Deprecated: GetValueFromIntOrPercent was deprecated in favor of
+//GetScaledValueFromIntOrPercent. This method was treating all int as a numeric value and all
+//strings with or without a percent symbol as a percentage value.
+func GetValueFromIntOrPercent(intOrPercent *IntOrString, total int, roundUp bool) (int, error) {
+	inlinedGetIntOrPercentValue := func(intOrStr *IntOrString) (int, bool, error) {
+		switch intOrStr.Type {
+		case Int:
+			return intOrStr.IntValue(), false, nil
+		case String:
+			s := strings.Replace(intOrStr.StrVal, "%", "", -1)
+			v, err := strconv.Atoi(s)
+			if err != nil {
+				return 0, false, fmt.Errorf("invalid value %q: %v", intOrStr.StrVal, err)
+			}
+			return int(v), true, nil
+		}
+		return 0, false, fmt.Errorf("invalid type: neither int nor percentage")
+	}
+
+	if intOrPercent == nil {
+		return 0, errors.New("nil value for IntOrString")
+	}
+	value, isPercent, err := inlinedGetIntOrPercentValue(intOrPercent)
+	if err != nil {
+		return 0, fmt.Errorf("invalid value for IntOrString: %v", err)
+	}
+	if isPercent {
+		if roundUp {
```

Comment:
This is kind of a horrifying block of code. There's actually (at least) four states people could want (floor, ceil, round, do nothing), and it's represented by a bool.

I recommend changing this function to return a float and let callers do the operation they need on this.

Reply:
(Just to be clear, I know you didn't write this!)

Reply:
This comment should be attached to the copy in the new function, of course :)